src/components/inputs.tsx

```diff
@@ -66,26 +66,41 @@ export const AmountInput = ({
 export const PercentageInput = ({
   ratio,
   onChange,
+  max,
 }: {
   ratio: Ratio;
   onChange: (newRatio: Ratio) => void;
+  max?: string;
 }) => {
   assert(
-    ratio.denominator.value === 10_000n,
-    'only conventional denominator value supported',
+    ratio.denominator.value <= 10_000n,
+    'Cannot handle denominators > 10_000n',
   );
+  const normalDenominator = 10_000n;
```

Comment:
should this be at module scope?

should definitely be before the `assert` to it can be used in it.
```js
assert(ratio.denominator.value <= normalDenominator, `Cannot handle denominators > ${normalDenominator`);
```

though… why can't it handle bigger denominators?

Reply:
Sure just moved it to module scope. 

Since bigints cannot be decimals, the scaleFactor would become 0 if it was greater than 10,000. Also technically anything not divisible by 10 would be scaled inaccurately, but I'm assuming we won't have to worry about that. The implementation just gets a lot more complicated if we have to generalize to handle any denominator.